src/preprocessing/point_in_poly/winding_number_jacobson.jl

```diff
@@ -0,0 +1,136 @@
+struct NaiveWinding end
+
+@inline function (winding::NaiveWinding)(mesh, query_point)
+    (; face_vertices) = mesh
+
+    return naive_winding(mesh, face_vertices, query_point)
+end
+
+@inline function naive_winding(mesh, faces, query_point)
+    winding_number = sum(faces, init=0.0) do face
+
+        # A. Van Oosterom 1983,
+        # The Solid Angle of a Plane Triangle (doi: 10.1109/TBME.1983.325207)
+        a = face_vertex(mesh, face, 1) - query_point
+        b = face_vertex(mesh, face, 2) - query_point
+        c = face_vertex(mesh, face, 3) - query_point
+        a_ = norm(a)
+        b_ = norm(b)
+        c_ = norm(c)
+
+        divisor = a_ * b_ * c_ + dot(a, b) * c_ + dot(b, c) * a_ + dot(c, a) * b_
+
+        return 2atan(det([a b c]), divisor)
```

Comment:
```suggestion
        return 2 * atan(det([a b c]) / divisor)
```
1. I find it easier to read with `2 * `
2. They just use a regular arctan in the paper, which is almost 2x faster.

Reply:
we need all quadrants (full range of `-pi < theta <= pi`) for a correct evaluation of the winding number.

Reply:
https://en.wikipedia.org/wiki/Atan2

Reply:
![image](https://github.com/user-attachments/assets/2b7004d4-5f21-49b7-be0b-fe77f6664bdd)